lib/browser/compression.js

```diff
@@ -29,15 +29,24 @@ async function deflate(method, value) {
 }
 
 function deflate_identity(value) {
-    return value;
+    if (Buffer.isBuffer(value)) {
```

Comment:
It would be more readable and maintainable to abstract this basically identical if/else from all these functions so it can be a one-liner, for example:
```javascript
return ensure_buffer_from_result(value);
```

And then you could just have
```javascript
export function buffer_from_result(result: value: ArrayBuffer | Buffer | Uint8Array): Buffer {
    if (Buffer.isBuffer(result)) {
        return result;
    } 
    return Buffer.from(result);
}